### ISSUE TITLE
[Bug]: @storybook/react-vite needs updated imports from `react-docgen` - exports were renamed in December 2022

### ISSUE THREAD

**Issue by nstuyvesant** [storybookjs/storybook#22406]
### Describe the bug

The module `react-docgen@6.0.0-rc.8` renamed several exports several commits ago in December of 2022:
https://github.com/reactjs/react-docgen/commit/0c2891d341796a05062276ab08ef1207de6770d6

This means that these lines of code will not work with the release candidate:
https://github.com/storybookjs/storybook/blob/next/code/frameworks/react-vite/src/plugins/react-docgen.ts#L5-L7

I will submit a PR.

**Comment by nstuyvesant** [comment]
Tried several things to resolve it...
1. Upgraded React on the package to 18 to match what Storybook is using. Thought there might be some conflict.
2. Turned off Storybook telemetry  in case that was interfering
3. Tried disabling the stories that are imported from core. Storybook has issues with MDX files coming from outside the package root (might need to eventually have a package.json script to just copy them, to the local package on builds).

None of those appeared to help.

**Comment by nstuyvesant** [comment]
If you look over the monorepo, you'll see that each package has a very similar approach to generating its storybook.

**Comment by nstuyvesant** [comment]
Could this have anything to do with...
```bash
yarn why react-docgen   
├─ @storybook/react-vite@npm:7.0.8
│  └─ react-docgen@npm:6.0.0-alpha.3 (via npm:6.0.0-alpha.3)
```

**Comment by nstuyvesant** [comment]
Wondering if this could be related... https://github.com/styleguidist/react-docgen-typescript/issues/481

**Comment by Gitsack** [comment]
Same issue with one of my repos - 7.0.7 worked fine. After upgrading to 7.0.8, there is just a white screen instead of storybook.

The following storybook packages are installed:

```
"@storybook/addon-a11y": "^7.0.7",
    "@storybook/addon-coverage": "^0.0.8",
    "@storybook/addon-essentials": "^7.0.7",
    "@storybook/addon-interactions": "^7.0.7",
    "@storybook/addon-links": "^7.0.7",
    "@storybook/addon-mdx-gfm": "^7.0.7",
    "@storybook/blocks": "^7.0.7",
    "@storybook/jest": "^0.1.0",
    "@storybook/nextjs": "^7.0.7",
    "@storybook/react": "^7.0.7",
    "@storybook/test-runner": "^0.10.0",
    "@storybook/testing-library": "^0.1.0",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^14.0.0",
```

Removing `addon-mdx-gfm` had no impact.

**Comment by nstuyvesant** [comment]
Here's the error I get and what it errored out on...
```
[Error] ReferenceError: Can't find variable: yd

	Module Code (assets/iframe-65deefa8.js:972:17829)
```

```js
yd.__docgenInfo = {
	description: '',
	methods: [
		{ name: 'addLineEventListener', docblock: null, modifiers: [], params: [], returns: null },
		{ name: 'addNodeEventListener', docblock: null, modifiers: [], params: [], returns: null },
		{
			name: 'traverse',
			docblock: null,
			modifiers: [],
			params: [
				{ name: 'e', type: null },
				{ name: 't', type: null },
				{ name: 'n', type: null }
			],
			returns: null
		},
		{ name: 'getRightArrowIcon', docblock: null, modifiers: [], params: [], returns: null },
		{ name: 'destroy', docblock: null, modifiers: [], params: [], returns: null }
	],
	displayName: 'yd'
}
```

It seems to be related to the `react-docgen` that Storybook uses.

**Comment by nstuyvesant** [comment]
I was able to get the site loading but leaving out the MDX files (which is not a suitable solution). See my update above.

**Comment by nstuyvesant** [comment]
When I forced the upgrade to `react-docgen@6.0.0-rc.8`, the error when I build the storybook is coming from
@storybook/react-vite/dist/preset.js:1:3525 which is this...
```js
defaultHandlers=Object.values(import_react_docgen.handlers).map(handler=>handler),defaultResolver=import_react_docgen.resolver.findAllExportedComponentDefinitions,defaultImporter=import_react_docgen.importers.makeFsImporter(),handlers=[...defaultHandlers,actualNameHandler]}});
```
That appears to be `@storybook/react-vite/src/plugins/react-docgen.ts:17` here: https://github.com/storybookjs/storybook/blob/next/code/frameworks/react-vite/src/plugins/react-docgen.ts#L16C1-L20

**Comment by shilman** [comment]
Is this actually a bug? Storybook depends on an older version of `react-docgen` and works fine with the version it depends on.

There are now a couple PRs to update to the latest RC: https://github.com/storybookjs/storybook/pull/22324 https://github.com/storybookjs/storybook/pull/22432 

We should unify these PRs. As far as I know this is a maintenance issue and would be released as part of 7.1. But if I'm missing something and it's a bug, I'm happy to try to get it in faster and patch it back to 7.0.x

Thanks for any clarification you can provide!
